src/node_interface.rs

```diff
@@ -373,6 +373,24 @@ impl NodeInterface {
             Err(NodeError::FailedParsingWalletStatus(res_json.pretty(2)))
         }
     }
+
+    /// Unlock wallet
+    pub fn wallet_unlock(&self, password: &String) -> Result<bool> {
```

Comment:
Few things since you mentioned you're new to rust:

- if we use `&str` for the parameter we don't need to convert it with `password.to_string()`, `&str` is also generally preferred over `&String` for func args taking string refs
- avoid panicking with unwraps, we should use the `?` operator instead and let library consumers handle errors
- No need to `clone` the body
- No need to parse the response if the status is success

What do you think about something like this?

```rust
    pub fn wallet_unlock(&self, password: &str) -> Result<bool> {
        let endpoint = "/wallet/unlock";
        let body = object! {
            pass: password,
        };

        let res = self.send_post_req(endpoint, body.to_string())?;

        if res.status().is_success() {
            Ok(true)
        } else {
            let json = self.parse_response_to_json(Ok(res))?;
            Err(NodeError::BadRequest(json["error"].to_string()))
        }
    }
```

Reply:
Thank you for the suggestion and detailed explanation. I agree with your proposed solution and will change my commit.